### PR TITLE
[#175476527] Release pipeline and pipeline refactor

### PR DESCRIPTION
**Pull request by balanza**
Same changes applied in https://github.com/pagopa/io-functions-services/pull/96.

### Changes in brief
* Introduce the `Release` stage on the pipeline which marks a release on the repository
* Remove `release-it` scripts and use plain `npm version` instead
* Refactor the whole CI/CD to have 
  * separated pipelines for code review and deploy scenarios
  * everything is into `.devops` folder now

### CODE REVIEW

**Comment by balanza** [comment]
Pipelines configured in Azure DevOps

File: .devops/deploy-pipelines.yml [hunk 1]
```diff
@@ -0,0 +1,166 @@
...
+    dependsOn:
+      - Deploy_staging
+    jobs:
+      - job: 'do_healthcheck'       
+        steps:  
+          - template: templates/rest-healthcheck/template.yaml@pagopaCommons 
+            parameters:
+              azureSubscription: '$(PRODUCTION_AZURE_SUBSCRIPTION)'
+              appName: '$(PRODUCTION_APP_NAME)'
+              endpoint: 'https://$(PRODUCTION_APP_NAME)-staging.azurewebsites.net/api/info'
```
Comment (gunzip) [inline .devops/deploy-pipelines.yml:145]:
```suggestion
              endpoint: 'https://$(PRODUCTION_APP_NAME)-staging.azurewebsites.net/api/v1/info'
```

**Comment by gunzip** [comment]
why draft?

File: .devops/deploy-pipelines.yml [hunk 2]
```diff
@@ -0,0 +1,168 @@
...
+                !.vscode/**/*
+                !azure-templates/**/*
+                !azure-pipelines.yml
+                !.prettierrc
+                !.gitignore
+                !README.md
+                !jest.config.js
+                !local.settings.json
+                !test
+                !tsconfig.json
```
Comment (pasqualedevita) [inline .devops/deploy-pipelines.yml:121]:
```suggestion
                **/*
                !.git/**/*
                !**/*.js.map
                !**/*.ts
                !.vscode/**/*
                !.devops/**/*
                !.prettierrc
                !.gitignore
                !README.md
                !jest.config.js
                !local.settings.json
                !test
                !tsconfig.json
                !tslint.json
                !yarn.lock
                !Dangerfile.js
                !CODEOWNERS
                !__*/**/*
```

added `!.devops/**/*` removed `!azure-templates/**/*`
others from fn-admin

**Review by pasqualedevita — CHANGES_REQUESTED** [review]
healthcheck ok

**Comment by gunzip** [comment]
can we merge this one?

**Pull request merged**